codex-rs/core/tests/summarize_context.rs

```diff
@@ -0,0 +1,109 @@
+#![expect(clippy::unwrap_used, clippy::expect_used)]
+
+//! Tests for the `Op::SummarizeContext` operation added to verify that
+//! summarization requests are properly handled and injected as user input.
+
+use std::time::Duration;
+
+use codex_core::Codex;
+use codex_core::protocol::EventMsg;
+use codex_core::protocol::InputItem;
+use codex_core::protocol::Op;
+mod test_support;
+use tempfile::TempDir;
+use test_support::load_default_config_for_test;
+use tokio::sync::Notify;
+use tokio::time::timeout;
+
+/// Helper function to set up a codex session and wait for it to be configured
+async fn setup_configured_codex_session() -> Codex {
+    let codex_home = TempDir::new().unwrap();
+    let config = load_default_config_for_test(&codex_home);
+    let (codex, _init_id) = Codex::spawn(config, std::sync::Arc::new(Notify::new()))
+        .await
+        .unwrap();
+
+    // Wait for session configured
+    loop {
+        let event = timeout(Duration::from_secs(5), codex.next_event())
+            .await
+            .expect("timeout waiting for session configured")
+            .expect("codex closed");
+
+        if matches!(event.msg, EventMsg::SessionConfigured(_)) {
+            break;
+        }
+    }
+
+    codex
+}
+
+#[tokio::test]
+async fn test_summarize_context_spawns_new_agent_task() {
+    // Test the specific behavior: when there's no current task,
+    // SummarizeContext should spawn a new AgentTask with the summarization prompt
+    let codex = setup_configured_codex_session().await;
+
+    // At this point, there should be no current task running
+    // Submit SummarizeContext operation - this should trigger:
+    // if let Err(items) = sess.inject_input(summarization_prompt) {
+    //     let task = AgentTask::spawn(Arc::clone(sess), sub.id, items);
+    //     sess.set_task(task);
+    // }
+    let _sub_id = codex.submit(Op::SummarizeContext).await.unwrap();
+
+    // Should receive a TaskStarted event indicating a new AgentTask was spawned
```

Comment:
I think this comment is duplicative with the message to your `assert!()`, so I would favor keeping the one in `assert!()` over this one.